pkg/scheduler/selector.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// subGroupSelector is a label selector to selector all pods in a sub podgroup
+type subGroupSelector []string
+
+func newSubGroupSelector(prefixGroups []string) labels.Selector {
+	return subGroupSelector(prefixGroups)
+}
+
+// Matches returns true if this selector matches the given set of labels.
+func (s subGroupSelector) Matches(labels labels.Labels) bool {
+	if !labels.Has(PodGroupLabel) {
+		return false
+	}
+
+	names := strings.Split(labels.Get(PodGroupLabel), subGroupSeparator)
+	if len(names) < len(s) {
+		return false
+	}
+
+	return joinNames(names[0:len(s)]) == joinNames(s)
+}
+
+// Empty returns true if this selector does not restrict the selection space.
+func (s subGroupSelector) Empty() bool {
+	return len(s) == 0
+}
+
+// String returns a human readable string that represents this selector.
+func (s subGroupSelector) String() string {
+	return "A label selector to selector all pods in a sub podgroup"
```

Comment:
I think this message lacks a detail: the podGroup name which is useful when debugging

Reply:
fixed